ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/PrintPublisher.java

```diff
@@ -112,8 +110,8 @@ public void subscribe(final Flow.Subscriber<Collection<String>> subscriber) {
           return null;
         }
 
-        final Collection<Supplier<String>> formatted = formatter.format(records);
-        final Collection<Supplier<String>> limited = new LimitIntervalCollection<>(
+        final Collection<String> formatted = formatter.format(records);
+        final Collection<String> limited = new LimitIntervalCollection<>(
             formatted,
             printTopic.getLimit().orElse(Integer.MAX_VALUE) - numWritten,
```

Comment:
This is that sets the batch size? So if it is not set then it is `(Integer.MAX_VALUE) - numWritten`  where `numWritten` what has been processed up to this point?

Reply:
If by batch size you're talking about the size of the batch of records pull from Kafka, then no.  That's controlled by the consumer config.

This code is only handling any `LIMIT` statement.